project_weather_app/data_handler.py

```diff
@@ -1,19 +1,26 @@
+"""
+The DataHandler directs the DataService 
+to get the data for a given location and 
+processes the data.
+"""
 from data_service import IDataService
 from location import Location
+import pandas as pd
+import matplotlib.pyplot as plt
 
 class DataHandler():
     """
     Given the data_service, get the data, and process it.
     """
     def __init__(self, data_service: IDataService):
         self.data_service = data_service
+        self.visualization_handler = VisualizationHandler()
```

Comment:
Here you should decouple the dependency and pass in the VisualizationHandler into the constructor rather than creating it within the constructor